assets/src/edit-story/components/form/color/colorPreview.js

```diff
@@ -161,35 +142,54 @@ function ColorPreview({
   ]);
 
   // Always hide color picker on unmount - note the double arrows
-  useEffect(() => () => hideSidebar(), [hideSidebar]);
-
-  if (isEditable) {
-    // If editable, only the visual preview component is a button
-    // And the text is an input field
-    return (
-      <Preview ref={ref}>
-        <VisualPreview role="status" style={previewStyle} {...buttonProps} />
-        <TextualInput
-          aria-label={`${inputLabel}: ${label}`}
-          value={hexInputValue}
-          onChange={handleInputChange}
-          onBlur={handleInputBlur}
-        />
-      </Preview>
-    );
-  }
+  useEffect(() => () => setPickerOpen(false), []);
 
-  // If not editable, the whole component is a button
   return (
-    <Preview ref={ref} {...buttonProps}>
-      <VisualPreview role="status" style={previewStyle} />
-      <TextualPreview>
-        {isMultiple
-          ? __('Multiple', 'web-stories')
-          : previewText ||
-            _x('None', 'No color or gradient selected', 'web-stories')}
-      </TextualPreview>
-    </Preview>
+    <>
+      {isEditable ? (
+        // If editable, only the visual preview component is a button
+        // And the text is an input field
+        <Preview ref={ref}>
+          <VisualPreview role="status" style={previewStyle} {...buttonProps} />
+          <TextualInput
+            aria-label={`${inputLabel}: ${label}`}
+            value={hexInputValue}
+            onChange={handleInputChange}
+            onBlur={handleInputBlur}
+          />
+        </Preview>
+      ) : (
+        // If not editable, the whole component is a button
+        <Preview ref={ref} {...buttonProps}>
+          <VisualPreview role="status" style={previewStyle} />
+          <TextualPreview>
+            {isMultiple
+              ? __('Multiple', 'web-stories')
+              : previewText ||
+                _x('None', 'No color or gradient selected', 'web-stories')}
+          </TextualPreview>
+        </Preview>
+      )}
+      <Popup
+        anchor={ref}
+        isOpen={pickerOpen}
+        placement={'left-start'}
+        spacing={{ x: 60 }}
+      >
+        <ColorPicker
+          color={isMultiple ? null : value}
+          onChange={onChange}
+          hasGradient={hasGradient}
+          hasOpacity={hasOpacity}
+          onClose={(evt) => {
+            setPickerOpen(false);
```

Comment:
This seems a bit inverse. I.e. I'd expect popup to be fully in charge of open/close and the content coordinates with that internally, which would involve some sort of context. But I think we can live with this for now. Just lmk your thoughts on this.